JavaScript/JavaScript.sublime-syntax

```diff
@@ -914,83 +902,65 @@ contexts:
 
   either-function-declaration:
     - match: '(?={{func_lookahead}})'
-      push: function-declaration
+      set: function-declaration
     - match: '(?={{arrow_func_lookahead}})'
-      push: arrow-function-declaration
+      set: arrow-function-declaration
 
   function-declaration:
-    - meta_scope: meta.function.declaration.js
-    - match: '\b(async)\b\s*'
-      captures:
-        1: storage.type.js
-    - match: \b(function)\b(\*)?\s*
-      captures:
-        1: storage.type.function.js
-        2: keyword.generator.asterisk.js
-    - match: '{{identifier}}'
-      scope: entity.name.function.js
-    - include: function-declaration-parameters
-    - match: '(?=\s*\{)'
-      set: function-block
+    - match: ''
+      set:
+        - - include: immediately-pop
+        - function-block
```

Comment:
Yeah, I would like that - but it isn't that way now so I agree it might be better to wait on that. I think you brought this up in https://github.com/sublimehq/Packages/issues/884 but as far as I know there isn't a definitive decision?